kbase-extension/static/kbase/js/widgets/kbaseTabs.js

```diff
@@ -241,45 +338,73 @@ define(['kbwidget', 'bootstrap', 'jquery', 'kbaseDeletePrompt'], (
             return this.data('tabs')[tabName];
         },
 
-        showTab: function (tab) {
-            this.data('nav')[tab].find('a').trigger('click');
+        showTab: function (tabName) {
+            this.data('nav')[tabName].find('a').trigger('click');
         },
 
         removeTab: function (tabName) {
             const $tab = this.data('tabs')[tabName];
             const $nav = this.data('nav')[tabName];
+            const deleteNav = () => {
+                $nav.remove();
+                delete this.data('nav')[tabName];
+            };
+
+            const deletePanel = () => {
+                $tab.remove();
+                delete this.data('tabs')[tabName];
+            };
+
+            const deleteTab = () => {
+                deleteNav();
+                deletePanel();
+            };
 
             if ($nav.hasClass('active')) {
-                if ($nav.next('li').length) {
-                    $nav.next().find('a').trigger('click');
+                const [$nextTab, $nextPanel] = (() => {
+                    if ($nav.next('li').length) {
+                        return [$nav.next('li'), $tab.next('[role="tabpanel"]')];
+                    } else if ($nav.prev('li').length) {
```

Comment:
No need for the `else` - can just do 
```js
if (condition_one) {
   return whatever;
}
if (condition_two) {
   return something_else;
}
return [null, null];
```

Reply:
Good catch, and I am a big fan of unraveling sequences of conditional branches into a sequence of if statements which just return, just got lazy here, I think, as I was transforming the original code and just adapted it. Although, to be fair, it is only stylistic as everyone understands a sequence of if-else. Still, changed.